### DISCUSSION TITLE
Boolean attributes as empty string values

### DISCUSSION THREAD

**Discussion by muratcorlu** [lit/lit#2814] [General]
I was trying to [solve an issue](https://stackoverflow.com/questions/71999979/storybook-web-component-code-examples-extra-comment-tags) I had in storybook, while documenting our web component library written with lit. As a summary;, for a template like this:

```html
html`<my-component ?disabled=${disabled}>Test</my-component>`;
```

I would expect a result 

```html
<my-component disabled>Test</my-component>
```

but actual result is 

```html
<my-component disabled="">Test</my-component>
```

First I thought this is a bug but then I realized that this is expected result in [lit-html tests](https://github.com/lit/lit/blob/main/packages/lit-html/src/test/lit-html_test.ts#L1131).

My question is, is there a specific reason to have `<tag disabled="">` instead of `<tag disabled>` as the result? If there is a requirement behind this decision I'll continue to look for a solution on the storybook side. Because I never want to suggest our library consumers to write empty string values for boolean attributes.

**Comment by justinfagnani** [comment]
This is just how boolean attributes behave. See [the `setAttribute()` docs on MDN](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute):

> Boolean attributes are considered to be true if they're present on the element at all. You should set value to the empty string ("") or the attribute's name, with no leading or trailing whitespace.

Even [`toggleAttribute(name, true)`](https://developer.mozilla.org/en-US/docs/Web/API/Element/toggleAttribute) sets the attribute to an empty string.

**Reply by muratcorlu** [reply]
OK. Then I’ll check my options to handle this in Storybook. Thank you for the clarification.
